### PR TITLE
Bug 1757216 Add ISP field to baseline_clients_daily

### PR DESCRIPTION
**Pull request by jklukas**
This will enable use to filter out clients sent by BrowserStack in downstream tables and queries.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql [hunk 1]
```diff
@@ -94,6 +95,7 @@ windowed AS (
     udf.mode_last(ARRAY_AGG(locale) OVER w1) AS locale,
     udf.mode_last(ARRAY_AGG(city) OVER w1) AS city,
     udf.mode_last(ARRAY_AGG(country) OVER w1) AS country,
+    udf.mode_last(ARRAY_AGG(isp) OVER w1) AS isp,
```
[the remark this answers is not in the document]
Comment (scholtzan) [reply sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql:98]: I don't think there are any prior cases for this. But I'd run `bqetl glean_usage generate` locally for the queries to get generated. I believe the `schema.yaml` files should get generated with them. Then a `bqetl query schema deploy 'moz-fx-data-shared-prod.*.baseline_clients_daily_v1'` should push the new schemas to BigQuery.

File: sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql [hunk 2]
```diff
@@ -33,6 +33,7 @@ WITH base AS (
     COALESCE(client_info.locale, metrics.string.glean_baseline_locale) AS locale,
     metadata.geo.city,
     metadata.geo.country,
+    metadata.geo.isp.`name` AS isp,
```
[the remark this answers is not in the document]
Comment (jklukas) [reply sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql:36]: It's `metadata.isp.name`; I incorrectly included it under `geo`. Will fix.

File: sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (jklukas) [reply sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql:98]: The schema files aren't automatically generated, but we can generate them via `bqetl query schema update`:

```
bqetl generate glean_usage
bqetl query schema update 'moz-fx-data-shared-prod.*.baseline_clients_daily_v1'
```

The above looks like it correctly gets the schema.yaml files staged, so I think we have a way forward. Thanks for the consult.

File: sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql [hunk 3]
```diff
@@ -33,6 +33,7 @@ WITH base AS (
     COALESCE(client_info.locale, metrics.string.glean_baseline_locale) AS locale,
     metadata.geo.city,
     metadata.geo.country,
+    metadata.isp.`name` AS isp,
```
Comment (jklukas) [inline sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql:36]: Fixed this path.

**Review by scholtzan — APPROVED** [review]

File: sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql (hunk 3 above)
Comment (jklukas) [reply sql_generators/glean_usage/templates/baseline_clients_daily_v1.query.sql:36]: `name` was being highlighted in my editor as a keyword, so it's probably a keyword for some SQL dialect. But it's not in [BigQuery's list of reserved keywords](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#reserved_keywords) so dropping the backticks should be fine.

**Comment by jklukas** [comment]
Closing this for now in favor of separate cleanup in https://github.com/mozilla/bigquery-etl/pull/2925. Will circle back to this question.

**Pull request closed without merging**